src/main/java/sirius/db/mixing/Property.java

```diff
@@ -716,12 +724,14 @@ protected final void onBeforeSave(Object entity) {
         Object propertyValue = getValue(entity);
         checkNullability(propertyValue);
 
-        if (propertyValidator != null) {
+        boolean valueChanged = entity instanceof BaseEntity<?> && (((BaseEntity<?>) entity).isNew()
+                                                                   || ((BaseEntity<?>) entity).isChanged(nameAsMapping));
+        if (propertyValidator != null && (strictValidation || valueChanged)) {
+            // Only enforce validity if the value actually changed or in strict validation mode
             propertyValidator.beforeSave(this, getValue(entity));
         }
 
-        if (entity instanceof BaseEntity<?> && (((BaseEntity<?>) entity).isNew() || ((BaseEntity<?>) entity).isChanged(
-                nameAsMapping))) {
+        if (valueChanged) {
             // Only enforce uniqueness if the value actually changed...
             checkUniqueness(entity, propertyValue);
```

Comment:
Maybe I'm blind, but uniqueness check should be called independent on validators (strict or not).

Reply:
It is called independent. There are two ifs, sharing the extracted valueChanged variable.